src/mesh/Router.cpp

```diff
@@ -239,6 +244,7 @@ ErrorCode Router::send(MeshPacket *p)
 #endif
 
         auto encodeResult = perhapsEncode(p);
+        if (isDirectMessage(p)) p->want_ack = true;
```

Comment:
I think isDirectMessage(p) will never be true if p->want_ack is not true already before, so this line will not do anything.